src/ru/nsu/ccfit/zuev/osu/menu/FilterMenu.java

```diff
@@ -22,8 +22,10 @@
 import ru.nsu.ccfit.zuev.osu.helper.InputManager;
 import ru.nsu.ccfit.zuev.osu.helper.StringTable;
 import ru.nsu.ccfit.zuev.osu.helper.TextButton;
+import ru.nsu.ccfit.zuev.osu.helper.DifficultyReCalculator;
 import ru.nsu.ccfit.zuev.osu.menu.SongMenu.SortOrder;
 import ru.nsu.ccfit.zuev.osuplus.R;
+import sun.util.resources.cldr.CalendarData;
 
 public class FilterMenu implements IUpdateHandler, IFilterMenu {
```

Comment:
If you set your filter to a newly added one (BPM, stars, or length), and then restart the game, the filter resets to title. Please add those new filters to `loadConfig` method, `saveConfig` method, and the switch statement in line 180 so that the aforementioned filters can be applied properly.

Reply:
I'll attempt to fix this, if I manage to do so, I'll comment the fixed code in this review.